### ISSUE TITLE
feat: support base64 strings in InputFile

### ISSUE THREAD

**Issue by KnorpelSenf** [grammyjs/grammY#705]
Title says it all. Should just be `new InputFile({ base64 })`.

**Comment by wojpawlik** [comment]
:-1:: https://github.com/tc39/proposal-arraybuffer-base64/issues/51

Why doesn't `fetchFile` check `response.ok`?

Next time I touch `InputFile`, I'd love to drop
- Deno-only `Deno.FsFile` -- no advantage over `new InputFile(fsFile.readable)`
- `URLLike` -- has to parse and construct `URL` each time: https://github.com/grammyjs/grammY/blob/ff4ebcbc2903a1e1e46c8e9b45ab253413f65caf/src/core/input_file.ts#L134

**Comment by KnorpelSenf** [comment]
> 👎: [tc39/proposal-arraybuffer-base64#51](https://github.com/tc39/proposal-arraybuffer-base64/issues/51)

Epic stuff, I expect it to land in Deno in a few weeks which is early enough for us to use it

> Why doesn't `fetchFile` check `response.ok`?

It should from now on.

> Next time I touch `InputFile`, I'd love to drop
> 
>     * Deno-only `Deno.FsFile` -- no advantage over `new InputFile(fsFile.readable)`

The advantage is that users don't have to think about how to convert a `Deno.FsFile` to something grammY understands. We can do this work for them.

>     * `URLLike` -- has to parse and construct `URL` each time: https://github.com/grammyjs/grammY/blob/ff4ebcbc2903a1e1e46c8e9b45ab253413f65caf/src/core/input_file.ts#L134

It does not have to do that, you just chose to do it. We can still change it.

**Comment by wojpawlik** [comment]
`Deno.FsFile` stays for now. It might need to be replaced with `{ readable: AsyncIterable<Uint8Array> }`.

`{ base64 }` (and `URLLike`) has to be parsed only once, in a function called from the constructor. Or in a static method.

What are non-goals for `InputFile`? What should it _not_ accept?
